docker_services_cli/cli.py

```diff
@@ -59,9 +47,55 @@ def add_env_output(ctx, *args, **kwargs):
                     env_set_command=env_set_command,
                 )
 
-        return update_wrapper(add_env_output, func)
+        return update_wrapper(_print_env_output, func)
+
+    return print_env_output
+
+
+def services_by_service_type(func):
+    """Add click options for each of the available service types."""
+
+    def collect_services_by_type(*args, **kwargs):
+        services = {}
+        for service_type in SERVICE_TYPES:
+            service = kwargs.pop(service_type)
+            if service:
+                services.setdefault(service_type, []).extend(
+                    service if isinstance(service, list) else [service]
+                )
+
+        kwargs["services"] = services
+        click.get_current_context().invoke(func, *args, **kwargs)
+
+    def validate_service_name(ctx, service_type, services_list):
+        available_services = SERVICE_TYPES.get(service_type.name, [])
+        for service in services_list:
+            if not (
+                normalize_service_name(service)
+                or normalize_service_name(service) in available_services
+            ):
+                raise click.BadParameter(
+                    "{service} is not a valid service of type {type}. "
+                    "Try one of: \n{available_services}".format(
+                        service=service, type=service_type.name,
+                        available_services=available_services
+                    )
+                )
+        return list(services_list)
+
+    for service_type in SERVICE_TYPES:
+        click.option(
+            "--{}".format(service_type),
+            callback=validate_service_name,
+            multiple=True,
```

Comment:
Question: Is it possible to pass more than one service of the same type? I might be wrong, but would something like this be allowed? `docker-services-cli up --db postgresql9 --db postgresql11`

I see that a [warning is triggered](https://github.com/inveniosoftware/docker-services-cli/pull/19/commits/8264b9d0acd005d6ab86de1ab7d817256d089861#diff-1fc38c9800c4622657c57dc615e70a5f0ff866314547364b8338fdef5a116d86R153-R156) when this happens and the user passes `--env`. My question is, should this be allowed? Do we need `multiple=True`?

This might make sense if we want to boot up, e.g. both MySQL and PostgreSQL databases, but in any case, `SQLALCHEMY_DATABASE_URI` gets overwritten so the former DB is not usable, am I right?

Reply:
Discussed offline: Keeping multiple options can be useful as a developer to try out different kinds of the same service, changing env vars manually.